railties/lib/rails/commands/clear/clear_command.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Rails
+  module Command
+    class ClearCommand < Base # :nodoc:
+      desc "clear", "Truncate log/*.log files"
+      option :logs, type: :array, lazy_default: [],
+        desc: "Truncate log/*.log files for specified environments to zero bytes. If no environments are specified, truncate logs for all environments."
+      def clear
+        if options[:logs].empty?
+          environments = all_environments
+        else
+          environments = options[:logs]
+        end
+        log_files_for(environments).each { |file| file.truncate(0) }
+      end
+
+      private
+        def log_files_for(environments)
+          pattern = environments == ["all"] ? "*" : "{#{environments.join(",")}}"
+          Rails::Command.application_root.glob("log/#{pattern}.log").select(&:file?)
+        end
+
+        def all_environments
+          Rails::Command.application_root.glob("config/environments/*.rb")
+            .map { |fname| File.basename(fname, ".*") }
+        end
```

Comment:
Since this is a brand new command, we *can* simplify the "empty" vs "all" distinction.  In other words, let's have `--logs` (with no values specified) truncate all `log/*.log`, and remove the special handling of "all".  (Sorry, I should have called more attention to that in https://github.com/rails/rails/pull/47623#issuecomment-1487364829.)

Reply:
@jonathanhefner `all` is slightly different as it will truncate any `log/*.log` instead of `log/{development,test,production}.log`
This was introduced in: https://github.com/rails/rails/issues/22544

Reply:
Yup. :smile:  I was saying that we can remove the old "empty" behavior, and keep the "all" behavior as the new "empty" behavior.

The idea is that:

* It's a bit odd to have `--logs all` behave differently than `--logs foo` or even `--logs ALL`.
* It makes more sense for bare `--logs` to affect every log that could possibly be specified via `--logs ...` (i.e. not limited to environments).
* Targeting a specific subset of logs is still possible via `--logs foo bar etc`.
* If the user doesn't want to clear all logs, it is likely that they want to clear a specific log.  For example, they want to clear `log/test.log` and don't actually care about clearing `log/production.log` (or vice versa).

@rafaelfranca  Do you still agree with keeping only the `LOGS=all` behavior?

Reply:
In this case we could remove the whole "checking if an environment even exists" part.
We might have to sanitize the arguments to prevent something like: `--logs ../.env/;`.

For the environments another alternative could be to use the --environment option that is used in other commands:
`bin/rails clear --logs --environment test`
Although that doesn't work with multiple environments.

Reply:
> If we allow passing any log to `--logs`, we could remove the whole "checking if an environment even exists" part.
> We might have to sanitize the arguments to prevent something like: `--logs ../.env/;`.

I might be misunderstanding, but there isn't a check in this PR nor in the Rake task.  And there probably doesn't need to be.  Consider the case where `log/production.log` doesn't exist yet, but a script invokes `bin/rails clear --logs production`.  I don't think that should cause the script to fail.

I also don't think we need to sanitize anything since we're not (or should not be) accepting untrusted input.

> For the environments another alternative could be to use the --environment option that is used in other commands:
> `bin/rails clear --logs --environment test`
> Although that doesn't work with multiple environments (maybe we could do `bin/rails clear --logs --environment local`).

That does not seem preferable to `bin/rails clear --logs test`.  Even `bin/rails clear --logs development test` is shorter than `bin/rails clear --logs --environment local` (39 characters vs 42 characters).  Also, we would still need to support `bin/rails clear --logs foo bar` for arbitrary non-environment log files.

> It would be more consistent with other commands and maybe it would be useful for `clear --tmp` as well.

How would it be useful for `clear --tmp`?  I don't think there are any files in `tmp/` that are segregated by environment, are there?

Reply:
> I might be misunderstanding, but there isn't a check in this PR nor in the Rake task. And there probably doesn't need to be. Consider the case where `log/production.log` doesn't exist yet, but a script invokes `bin/rails clear --logs production`. I don't think that should cause the script to fail.

Ah sorry, my mistake. I thought we'd do `log/*.log` or filter input on one of the environments using the following:

https://github.com/rails/rails/blob/6879f89892b4784123fb40cdcc983ac32590a24d/railties/lib/rails/tasks/log.rake#L38-L40

I wasn't aware we can currently do:
```bash
bin/rails log:clear ENV['LOGS']='dummy,puma,redis'
```

> That does not seem preferable to `bin/rails clear --logs test`. Even `bin/rails clear --logs development test` is shorter than `bin/rails clear --logs --environment local` (39 characters vs 42 characters). Also, we would still need to support `bin/rails clear --logs foo bar` for arbitrary non-environment log files.

I think I agree that is preferable.
Also having `--environment` accept multiple environments is inconsistent with other commands.

> How would it be useful for `clear --tmp`? I don't think there are any files in `tmp/` that are segregated by environment, are there?

Hmm, I seem to recall Rails used to cache assets separately in `tmp` but I might be confusing that with the `public/packs` and `public/packs-test` directories.

I do wonder if `assets:clean` should be handled by `clear` as well.
So `clear` would basically clear **all** generated files.
But that is of later concern.